src/dmd/dtemplate.d

```diff
@@ -8173,3 +8173,58 @@ MATCH matchArg(TemplateParameter tp, Scope* sc, RootObject oarg, size_t i, Templ
     else
         assert(0);
 }
+
+
+/***********************************************
+ * Collect and print statistics on template instantiations.
+ */
+struct TemplateStats
+{
+    __gshared TemplateStats[const void*] stats;
+
+    uint numInstantiations;     // number of instantiations of the template
+    uint uniqueInstantiations;  // number of unique instantiations of the template
+
+    /*******************************
+     * Add this instance
+     */
+    static void incInstance(const TemplateDeclaration td)
+    {
+        if (!global.params.vtemplates)
+            return;
+        if (!td)
+            return;
+        if (auto ts = cast(const void*) td in stats)
+            ++ts.numInstantiations;
+        else
+            stats[cast(const void*) td] = TemplateStats(1, 0);
+    }
+
+    /*******************************
+     * Add this unique instance
+     */
+    static void incUnique(const TemplateDeclaration td)
+    {
+        if (!global.params.vtemplates)
+            return;
+        if (!td)
+            return;
+        if (auto ts = cast(const void*) td in stats)
+            ++ts.uniqueInstantiations;
+        else
+            stats[cast(const void*) td] = TemplateStats(0, 1);
+    }
+}
+
+
+void printTemplateStats()
+{
+    if (!global.params.vtemplates)
+        return;
+
+    printf("  Number   Unique   Name\n");
+    foreach (td, ref ts; TemplateStats.stats)
+    {
+        printf("%8u %8u   %s\n", ts.numInstantiations, ts.uniqueInstantiations, (cast(const TemplateDeclaration) td).toChars());
```

Comment:
This should propably print the fully-qualified name of the template, your [example in the forum](https://forum.dlang.org/post/rb724g$2aum$1@digitalmars.com) lists multiple `Impl(T)` templates

Reply:
I considered it but decided against it to keep the column length down so it doesn't wrap so much. The current method also makes the most important part of the name come first.

Reply:
Fair enough, maybe append file+line instead?

Reply:
How about just pulling it, and if it turns out to be an actual issue we can revisit it then. I designed it so it should be easy to add more stuff to.

Reply:
I would like to support showing this possibly annotated with an IDE. Can we add a switch to print the fully qualified name to simplify symbol lookup?

Reply:
the fqn of a deep template instance is not something you want to see.
it would take ages printing as well

Reply:
ok I guess the full name isn't the most important. Having file+line as suggested by MoonlightSentinel would actually cover the use case I had thought of (would have looked up file by module name and grepped for the definition with the full template name)